lib/src/lints/flutter/proper_controller_dispose.dart

```diff
@@ -120,4 +90,136 @@ class ProperControllerDispose extends DartLintRule {
       }
     });
   }
+
+  @override
+  List<Fix> getFixes() => [_ProperControllerDisposeFix()];
+}
+
+Iterable<VariableDeclaration> _getControllerDeclarations(
+  Iterable<FieldDeclaration> fieldDeclarations,
+) {
+  return fieldDeclarations
+      .where((e) {
+        final type = e.fields.type?.type;
+        if (type != null) {
+          if (disposableControllerChecker.isExactlyType(type) ||
+              changeNotifierChecker.isSuperTypeOf(type)) {
+            return true;
+          }
+        }
+
+        if (type == null) {
+          final variableDeclarations = e.fields.variables;
+          if (variableDeclarations.length == 1) {
+            final variableDeclaration = variableDeclarations.first;
+            final initializer = variableDeclaration.initializer;
+
+            if (initializer is InstanceCreationExpression) {
+              final staticType = initializer.staticType;
+              return switch (staticType) {
+                null => false,
+                final type => disposableControllerChecker.isExactlyType(type) ||
+                    changeNotifierChecker.isSuperTypeOf(type),
+              };
+            }
+          }
+        }
+
+        return false;
+      })
+      .map((e) => e.fields.variables)
+      .expand((e) => e);
+}
+
+class _ProperControllerDisposeFix extends DartFix {
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    AnalysisError analysisError,
+    List<AnalysisError> others,
+  ) {
+    context.registry.addClassDeclaration((node) {
+      if (!analysisError.sourceRange.intersects(node.sourceRange)) return;
+
+      final type = node.extendsClause?.superclass.type;
+      if (type == null || !widgetStateChecker.isAssignableFromType(type)) {
+        return;
+      }
+
+      final fieldDeclarations = node.members.fieldDeclarations;
+      final controllerDeclarations =
+          _getControllerDeclarations(fieldDeclarations);
+      final disposeBlock = node.members.findMethodDeclarationByName('dispose');
+
+      final changeBuilder = reporter.createChangeBuilder(
+        message: 'Dispose all the controllers',
+        priority: 80,
+      );
+
+      if (disposeBlock == null) {
+        int? offset;
+        offset = node.members.findMethodDeclarationByName('initState')?.end;
+        offset = offset != null ? offset + 2 : null;
+
+        offset =
+            offset ?? node.members.findMethodDeclarationByName('build')?.offset;
+        offset = offset != null ? offset - 1 : null;
+
+        offset = offset ?? node.end - 2;
+
+        final disposeList = controllerDeclarations
+            .map((e) => '${e.name.lexeme}.dispose();')
+            .join('\n');
+
+        changeBuilder.addDartFileEdit((builder) {
+          builder.addSimpleInsertion(offset!, '''
+
+@override
+void dispose() {
+$disposeList
+super.dispose();
+} 
+ ''');
```

Comment:
The logic here is a bit hard to follow.

I would probably to something like the below and use `builder.addInsertion()` instead

```dart
// This is just a draft
final initStateMethod = node.members.findMethodDeclarationByName(
  'initState',
);

final buildMethod = node.members.findMethodDeclarationByName('build');

final (int offset, bool addNewLineAtTheStart, bool addNewLineAtTheEnd) =
    switch ((initStateMethod, buildMethod)) {
  (null, null) => (node.end, true, false),
  (null, final buildMethod?) => (buildMethod.offset, true, true),
  (final initStateMethod?, null) => (initStateMethod.end, true, false),
  (final _?, final buildMethod?) => (buildMethod.offset, false, true),
};

changeBuilder.addDartFileEdit((builder) {
  builder.addInsertion(offset, (builder) {
    if (addNewLineAtTheStart) builder.write('\n');
    builder.write('  @override\n');
    builder.write('  void dispose() {\n');
    for (final c in controllerDeclarations) {
      builder.write('    ${c.name.lexeme}.dispose();\n');
    }
    builder.write('    super.dispose();\n');
    builder.write('  }\n');
    if (addNewLineAtTheEnd) builder.write('\n');
  });
});
```